kuflow-robotframework/KuFlow/keywords.py

```diff
@@ -35,31 +35,46 @@
 
 from kuflow_rest import KuFlowRestClient, models
 
-# from models import Pr
-
 
 class Keywords:
     def __init__(self) -> None:
         self.logger = logging.getLogger(__name__)
         self._client = None
 
     @keyword(tags=("settings",))
-    def set_client_authentication(self, endpoint, client_id, client_secret):
+    def set_client_authentication(self, client_id: str, client_secret: str, endpoint: Optional[str] = None):
         """Configure the client authentication in order to execute keywords against Rest API.
 
         Before using any other KuFlow Keyword, this one must be called.
 
         Example:
-        | Set Client Authentication | %{KUFLOW_API_ENDPOINT} | %{KUFLOW_CLIENT_ID} | %{KUFLOW_CLIENT_SECRET}
+        | Set Client Authentication | %{KUFLOW_CLIENT_ID} | %{KUFLOW_CLIENT_SECRET}
+        | Set Client Authentication | %{KUFLOW_CLIENT_ID} | %{KUFLOW_CLIENT_SECRET} | %{KUFLOW_API_ENDPOINT}
         =>
-        | Set Client Authentication | https://api.kuflow.com/v1.0 | identifier | token
+        | Set Client Authentication | identifier | token
+        | Set Client Authentication | identifier | token | https://api.kuflow.com/v1.0
         """
-        self._client = KuFlowRestClient(
-            client_id=client_id,
-            client_secret=client_secret,
-            endpoint=endpoint,
-            allow_insecure_connection=True,
-        )
+        if endpoint is None:
```

Comment:
Can we handle this inside KuFlowRestClient?

Reply:
Is it already inside KuFlowRestClient. However, this here is to make syntactic sugar for the RobotFramework keyword. Because I want to avoid putting in the definition of this method:
`[XXXXXXXXXXXXXX](endpoint: str = "https://api.kuflow.com/v2022-10-08")`
because it is already inside. And I want to leave as optional (because only we use it) to have to pass the endpoint in the keyword. So this is the only way I can think of. It is difficult to explain, if you want I will show you with an example.